python-sdk/src/astro/table.py

```diff
@@ -129,10 +129,15 @@ def name(self, value: str) -> None:
     @property
     def row_count(self) -> Any:
         """
-        Return the row count of table
+        Return the row count of table.
         """
-        # TODO: Implement this property
-        return 0
+        try:
+            db = create_database(self.conn_id)
+            return db.run_sql(
+                f"SELECT COUNT(*) from {db.get_table_qualified_name(self)}"  # skipcq: BAN-B608
+            ).scalar()
+        except Exception:
+            return 0
```

Comment:
Not sure if it’s a good idea to blanket-catch and return 0 here; why not let the exception bubble up?

Reply:
@uranusjr If I do Table(...) then it just creates metadata and not the actual table in the source database because of that if I do Table(...).row_count it would complain table does not exist but yeah agree it is not a good idea to ignore the exception.

Reply:
It should be possible to do some reflection to check whether the table exists. It is not unreasonable to simply raise an exception in this case either.

Reply:
removed try-catch block